src/Amadevus.RecordGenerator.Attributes/Features.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+
+namespace Amadevus.RecordGenerator
+{
+    /// <summary>
+    /// Specifies record features that should be generated.
+    /// </summary>
+    [Flags]
+    public enum Features
+    {
+        /// <summary>
+        /// Zero value.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// Record constructor that takes and assigns all read-only properties.
+        /// </summary>
+        Constructor = 0b_1,
+
+        /// <summary>
+        /// With* method per property that creates new record instance with
+        /// provided value assigned to the associated property. Also includes
+        /// generation of Update method.
+        /// </summary>
+        WithPerProperty = 0b_10,
```

Comment:
This definitely needs a better and simpler name. I was thinking `Mutators` but that can give the wrong impression of in-place update and mutability when it's copy-and-update in actual fact. I don't have a good suggestion for now but will drop with one if it pops in my mind later.

Reply:
At first I named it just `With`. But I've always wanted to somehow replace those separate methods with a single `With` method that'd take all parameters with default values of some generic wrapper so that those unspecified could be replaced with current values.

```csharp
public MyRecord With(
    Maybe<string> id,
    Maybe<int?> size,
    Maybe<MyRecord> inner)
{
    return Update(id.IfNone(Id), size.IfNone(Size), inner.IfNone(Inner);
}
```

That's why I didn't keep simple `With`. Maybe `WithTransforms`? I don't know.

Reply:
@jflepp @atifaziz Let's do a quick vote taking into account the above:

Name | Reaction
-|-
WithPerProperty | 👀
Withers | 🚀
With | 👍
WithTransforms | 😄
Mutators | 🎉

You can vote for multiple.

Reply:
I just like Mutators because the missing With could lead to confusion and the WithTransforms because "Transform" sounds more like a change of state to me.

Reply:
> Let's do a quick vote

Interesting way to conduct a poll. 😄

It would have been nice to have the option to vote, “(something else)”.

Reply:
You're welcome to propose other options, but I want to close this PR with *something*, probably something that's not WithPerProperty :)

Reply:
@amis92 `CopyModifiers`? That's the best I can come up with. Perhaps I am trying too hard to come up with a _logical_ feature name but I think what kept throwing me off about `Withers` (while creative) is that it also implies `Update`. Actually, the [C# record design document](https://github.com/dotnet/csharplang/blob/master/proposals/records.md) currently [states](https://github.com/dotnet/csharplang/blob/master/proposals/records.md#with-method):

> ##### `With` method
>
> Unless there is a user-declared member named `With` declared, a record type has a compiler-provided method named `With` whose return type is the record type itself, and containing one value parameter corresponding to each *record-parameter* in the same order that these parameters appear in the record type declaration. Each parameter shall have a *caller-receiver default-argument* of the corresponding property.

This sounds awfully like `Update` (minus the defaulted arguments) so I wonder if we should align with the above and at which point, we'll have only _withers_!

Reply:
What about `WithMethods`?
Because it is being talked about a "method named `With`"

In order to be consequent with the names we could also use `BuilderClass` and `ToStringMethod()`

Reply:
I think I'll go with `Withers`, since just `With` might be confusing, and `Withers` stands apart enough for a person to know what's that or check the doc comment.

Reply:
I think that's a fair call for now. Any reason to not have _withers_ in the default set?

Reply:
Haha, okay that's weird. My last comment "I think I'll go with..." was written because GitHub didn't update my page with new comments and I thought no-one responded with new proposals. Oh my oh my.

So. Appending member type to names *is* an interesting option, but it kind of feels incorrect for me because we're talking about Features, and some features like the incoming Equality will contain more members/different names than specified. `CopyModifiers` sounds too close to modifiers in syntax terms, like `public, private, internal` etc.

Essentially, the WithXyz+Update is what Roslyn implemented for their `SyntaxNode`s. It also kind of helps with creating record hierarchies, because the `Update` is then overridden in inherited class and even if you do `Base.WithA` it calls Update which resolves to `Inherited.Update` that returns new `Inherited`. That's why Update even exists, actually.

As declared, I'll merge this for now with the `Withers` feature name, but we can still continue discussing that in #45. I'm still not really satisfied with `Withers` as is. But we're still in semantic version `0` so changes like that should be easy (also changing the enum name won't break anything on the tooling side because we resolve to the int value anyway).